### PR TITLE
feat: vulnerability location

### PR DESCRIPTION
**Pull request by SmileOstorlab**

### CODE REVIEW

File: agent/nmap_agent.py [hunk 1]
```diff
@@ -141,9 +142,24 @@ def _emit_network_scan_finding(self, scan_results: Dict[str, Any], normal_result
         scan_result_technical_detail = process_scans.get_technical_details(scan_results)
         if normal_results is not None:
             technical_detail = f'{scan_result_technical_detail}\n```xml\n{normal_results}\n```'
-            self.report_vulnerability(entry=kb.KB.NETWORK_PORT_SCAN,
-                                      technical_detail=technical_detail,
-                                      risk_rating=vuln_mixin.RiskRating.INFO)
+            port = scan_results.get('nmaprun', {}).get('host', {}).get('@portid', '')
+            res = scan_results.get('nmaprun', {}).get('host', {}).get('address', '')
```
Comment (deadly-panda) [inline agent/nmap_agent.py:146]: The name res is confusing

File: agent/nmap_agent.py [hunk 2]
```diff
@@ -141,9 +142,24 @@ def _emit_network_scan_finding(self, scan_results: Dict[str, Any], normal_result
...
+            if res.get('@addrtype', '') == 'ipv4':
+                self.report_vulnerability(entry=kb.KB.NETWORK_PORT_SCAN,
+                                          technical_detail=technical_detail,
+                                          risk_rating=vuln_mixin.RiskRating.INFO,
+                                          vulnerability_location=vuln_mixin.VulnerabilityLocation(
+                                              metadata=[],
```
Comment (deadly-panda) [inline agent/nmap_agent.py:152]: If you have the port add it in the meta data.

File: agent/nmap_agent.py [hunk 3]
```diff
@@ -141,9 +142,24 @@ def _emit_network_scan_finding(self, scan_results: Dict[str, Any], normal_result
...
+                                              asset=ipv4.IPv4(host=res.get('@addr', ''), mask=port)
```
Comment (deadly-panda) [inline agent/nmap_agent.py:153]: mask=port ?

File: agent/nmap_agent.py [hunk 4]
```diff
@@ -141,9 +142,24 @@ def _emit_network_scan_finding(self, scan_results: Dict[str, Any], normal_result
...
+                                          ))
+            else:
```
Comment (deadly-panda) [inline agent/nmap_agent.py:155]: elif ipv6

Comment (deadly-panda) [inline agent/nmap_agent.py:155]: I don't see the case for the domain target ? In that case you should emit the hostname as a domain asset.

**Review by deadly-panda — DISMISSED** [review]
Remember to add unit tests.

File: agent/nmap_agent.py (hunk 1 above)
Comment (3asm) [inline agent/nmap_agent.py:146]: res for addres?

File: agent/nmap_agent.py [hunk 5]
```diff
@@ -141,9 +142,24 @@ def _emit_network_scan_finding(self, scan_results: Dict[str, Any], normal_result
         scan_result_technical_detail = process_scans.get_technical_details(scan_results)
         if normal_results is not None:
             technical_detail = f'{scan_result_technical_detail}\n```xml\n{normal_results}\n```'
-            self.report_vulnerability(entry=kb.KB.NETWORK_PORT_SCAN,
-                                      technical_detail=technical_detail,
-                                      risk_rating=vuln_mixin.RiskRating.INFO)
+            port = scan_results.get('nmaprun', {}).get('host', {}).get('@portid', '')
+            res = scan_results.get('nmaprun', {}).get('host', {}).get('address', '')
+            if res.get('@addrtype', '') == 'ipv4':
```
Comment (3asm) [inline agent/nmap_agent.py:147]: worth putting in a variable.

File: agent/nmap_agent.py (hunk 2 above)
Comment (3asm) [inline agent/nmap_agent.py:152]: metadata is empty!

File: agent/nmap_agent.py [hunk 6]
```diff
@@ -141,9 +142,24 @@ def _emit_network_scan_finding(self, scan_results: Dict[str, Any], normal_result
...
+                                          vulnerability_location=vuln_mixin.VulnerabilityLocation(
+                                              metadata=[],
+                                              asset=ipv4.IPv4(host=res.get('@addr', ''), mask=port)
+                                          ))
+            else:
+                self.report_vulnerability(entry=kb.KB.NETWORK_PORT_SCAN,
+                                          technical_detail=technical_detail,
+                                          risk_rating=vuln_mixin.RiskRating.INFO,
+                                          vulnerability_location=vuln_mixin.VulnerabilityLocation(
+                                              metadata=[],
```
Comment (3asm) [inline agent/nmap_agent.py:160]: metadata is also empty.

File: agent/nmap_agent.py [hunk 7]
```diff
@@ -141,9 +142,24 @@ def _emit_network_scan_finding(self, scan_results: Dict[str, Any], normal_result
...
+                                              asset=ipv6.IPv6(host=res.get('@addr', ''), mask=port)
+                                          ))
```
Comment (3asm) [inline agent/nmap_agent.py:162]: NO UNIT TEST!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Review by 3asm — CHANGES_REQUESTED** [review]

File: agent/nmap_agent.py (hunk 7 above)
Comment (3asm) [reply agent/nmap_agent.py:180]: I still don't see any unit test checking for the values in vulnerability location.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: agent/nmap_agent.py [hunk 8]
```diff
@@ -13,6 +13,7 @@
 from ostorlab.agent.message import message as msg
 from ostorlab.agent.mixins import agent_persist_mixin as persist_mixin
 from ostorlab.agent.mixins import agent_report_vulnerability_mixin as vuln_mixin
+from ostorlab.assets import ipv4, ipv6, domain_name
```
Comment (3asm) [inline agent/nmap_agent.py:16]: I think it would be cleaner to use domain_name_asset and keep the variable name as is.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**